### PR TITLE
feat: liquidity mining for xyk pallet

### PR DESCRIPTION
**Pull request by martinfridrich**
## Description
This PR adds liquidity mining support for xyk pools.

### CODE REVIEW

File: pallets/xyk-liquidity-mining/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,964 @@
...
+			min_deposit: Balance,
+			price_adjustment: FixedU128,
+		) -> DispatchResult {
+			<T as pallet::Config>::CreateOrigin::ensure_origin(origin)?;
+
+			if !T::AssetRegistry::is_sufficient(reward_currency) {
+				let ed =
+					T::AssetRegistry::existential_deposit(reward_currency).ok_or(Error::<T>::AssetNotRegistered)?;
+
+				let pot = T::LiquidityMiningHandler::pot_account().ok_or(Error::<T>::FailToGetPotId)?;
```
Comment (jak-pan) [inline pallets/xyk-liquidity-mining/src/lib.rs:365]: As I understand we need to seed the pot with ED on farm creation. We have two pots though, the first one is seeded on transfer by the farm creator. We could put some comment about this fact.

File: pallets/xyk-liquidity-mining/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,964 @@
...
+		type AMM: AMM<Self::AccountId, AssetId, AssetPair, Balance>
+			+ AMMPosition<AssetId, Balance, Error = DispatchError>;
+
+		/// The origin account that can create new liquidity mining program.
+		type CreateOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// Pallet id.
+		type PalletId: Get<PalletId>;
+
+		/// NFT collection id for liq. mining deposit nfts. Has to be within the range of reserved NFT class IDs.
```
Comment (jak-pan) [inline pallets/xyk-liquidity-mining/src/lib.rs:126]:
```suggestion
```

**Review by jak-pan — COMMENTED** [review]
Just small stuff otherwise LGTM. 
Did we do benchmarks yet?

**Comment by martinfridrich** [comment]
> Just small stuff otherwise LGTM. Did we do benchmarks yet?

Yes,  committed weights are from new benchmarks with insufficient assets

**Review by jak-pan — APPROVED** [review]

**Review by mrq1911 — APPROVED** [review]

**Pull request merged**